src/Microsoft.ML.Data/Data/DataViewUtils.cs

```diff
@@ -402,7 +402,7 @@ private static DataViewRowCursor ConsolidateCore(IChannelProvider provider, Data
                                             // The waiter event should never be null since this is only
                                             // called after a point where waiter.Register has been called.
                                             ch.AssertValue(waiterEvent);
-                                        waiterEvent.Wait();
+                                        waiterEvent.Wait(5*60*1000);
```

Comment:
📝 This change should be reverted for the same reason I mentioned for **OrderedWaiter.cs**.